src/main/java/com/icthh/xm/ms/dashboard/mapper/DashboardMapper.java

```diff
@@ -0,0 +1,92 @@
+package com.icthh.xm.ms.dashboard.mapper;
+
+import com.icthh.xm.ms.dashboard.domain.Dashboard;
+import com.icthh.xm.ms.dashboard.domain.Widget;
+import com.icthh.xm.ms.dashboard.service.dto.DashboardDto;
+import com.icthh.xm.ms.dashboard.service.dto.WidgetDto;
+import org.springframework.stereotype.Component;
+
+import java.util.Set;
+
+import static java.util.stream.Collectors.toSet;
+
+@Component
+public class DashboardMapper {
```

Comment:
Please consider of using MapStruct as proposed by Jhipster:
https://www.jhipster.tech/using-dtos/

Reply:
Thanks for the suggestion. The current approach is easier to read understand and build.  Any king of dynamic mapping or metadata based mappers works for Hello world +1 word dtos when you have complex dtos which are quite possible to have if your application is still in development and still accepts any kind of changes. So I prefer simple java classes!